modules/payloads/adapters/cmd/unix/python.rb

```diff
@@ -0,0 +1,42 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+module MetasploitModule
+  include Msf::Payload::Adapter
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Python Exec',
+        'Description' => 'Execute a Python payload from a command.',
+        'Author' => 'Spencer McIntyre',
+        'Platform' => 'unix',
+        'Arch' => ARCH_CMD,
+        'License' => MSF_LICENSE,
+        'AdaptedArch' => ARCH_PYTHON,
+        'AdaptedPlatform' => 'python'
+      )
+    )
+  end
+
+  def compatible?(mod)
+    if mod.type == Msf::MODULE_PAYLOAD && (mod.class.const_defined?(:CachedSize) && mod.class::CachedSize != :dynamic) && (mod.class::CachedSize >= 120_000) # echo does not have an unlimited amount of space
```

Comment:
Just confirming, will this size check work if there's heavy amounts of escaping required as part of the generate command below? 🤔 

```
Shellwords.escape(payload)
```

Reply:
Well the generated payloads *should* all have gone through `Msf::Payload::Python.create_exec_stub` to have been squashed into a single express, with the bulk of the contents in the base64 blob. That means the number of characters that need to be encoded is relatively consistent regardless of the contents. Appears to be about 24 bytes. If a payload isn't going through `create_exec_stub` we should probably update it to, so all Python payloads are a single one-line expression. I know at this point some are not like [python/pingback_reverse_tcp](https://github.com/rapid7/metasploit-framework/blob/da044a967a39816f25c6978c13ca655dea6f4c89/modules/payloads/singles/python/pingback_reverse_tcp.rb#L30) which is why there's a check for newline characters below this to squash those ones into a single expression.

```
[3] pry(#<#<Class:0x00007f07a4a3f978>>)> exec_stub = Msf::Payload::Python.create_exec_stub('hello'); Shellwords.escape(exec_stub).length - exec_stub.length
=> 23
[4] pry(#<#<Class:0x00007f07a4a3f978>>)> exec_stub = Msf::Payload::Python.create_exec_stub("'" * 1024); Shellwords.escape(exec_stub).length - exec_stub.length

=> 24
[5] pry(#<#<Class:0x00007f07a4a3f978>>)> exec_stub = Msf::Payload::Python.create_exec_stub("\\" * 1024); Shellwords.escape(exec_stub).length - exec_stub.length

=> 24
[6] pry(#<#<Class:0x00007f07a4a3f978>>)>
```